nntrainer/utils/base_properties.h

```diff
@@ -133,28 +160,44 @@ template <typename T> class Property {
    *
    * @return T value
    */
-  operator T &() { return value; }
+  operator T &() { return get(); }
 
   /**
    * @brief cast operator for property
    *
    * @return T value
    */
-  operator const T &() const { return value; }
+  operator const T &() const { return get(); }
```

Comment:
T operator? operator T?

Reply:
Operator T & is a casting operator.
The intention here is to make using property as if it's underlying value type :)

```cpp
const auto name = property<string>();
std::string& name_ = name; ///this function is called
```